spec/src/consensus.rs

```diff
@@ -132,24 +132,87 @@ impl Default for ConsensusBuilder {
             .input(input)
             .witness(witness)
             .build();
-        let dao = genesis_dao_data_with_satoshi_gift(
+        let difficulty = U256::one();
+        let epoch_ext = build_genesis_epoch_ext(DEFAULT_EPOCH_REWARD, &difficulty);
+
+        let dao = build_genesis_dao_data(
             vec![&cellbase],
             &SATOSHI_PUBKEY_HASH,
             SATOSHI_CELL_OCCUPIED_RATIO,
-        )
-        .unwrap();
+            &epoch_ext,
+            TX_PROPOSAL_WINDOW,
+            DEFAULT_SECONDARY_EPOCH_REWARD,
+        );
         let genesis_block = BlockBuilder::default()
-            .difficulty(U256::one().pack())
+            .difficulty(difficulty.pack())
             .dao(dao)
             .transaction(cellbase)
             .build();
 
-        ConsensusBuilder::new(genesis_block, DEFAULT_EPOCH_REWARD)
+        ConsensusBuilder::new(genesis_block, DEFAULT_EPOCH_REWARD, epoch_ext)
     }
 }
 
+pub fn build_genesis_epoch_ext(epoch_reward: Capacity, difficulty: &U256) -> EpochExt {
+    let block_reward = Capacity::shannons(epoch_reward.as_u64() / GENESIS_EPOCH_LENGTH);
+    let remainder_reward = Capacity::shannons(epoch_reward.as_u64() % GENESIS_EPOCH_LENGTH);
+
+    let genesis_hash_rate =
+        difficulty * (GENESIS_EPOCH_LENGTH + GENESIS_ORPHAN_COUNT) / EPOCH_DURATION_TARGET;
+
+    EpochExt::new_builder()
+        .number(0)
+        .base_block_reward(block_reward)
+        .remainder_reward(remainder_reward)
+        .previous_epoch_hash_rate(genesis_hash_rate)
+        .last_block_hash_in_previous_epoch(Byte32::zero())
+        .start_number(0)
+        .length(GENESIS_EPOCH_LENGTH)
+        .difficulty(difficulty.clone())
+        .build()
+}
+
+pub fn build_genesis_dao_data(
+    txs: Vec<&TransactionView>,
+    satoshi_pubkey_hash: &H160,
+    satoshi_cell_occupied_ratio: Ratio,
+    genesis_epoch_ext: &EpochExt,
+    tx_proposal_window: ProposalWindow,
+    secondary_epoch_reward: Capacity,
+) -> Byte32 {
+    // Genesis block has special issurance rule, hence we only add up
+    // issurance from block 1 up to the length of proposal window here.
+    let initial_primary_issurance = genesis_epoch_ext
+        .block_reward(0)
+        .and_then(|c| {
+            c.safe_mul(tx_proposal_window.farthest())
+                .map_err(Into::into)
+        })
+        .expect("initial primary issurance calculation overflow!");
+    let initial_secondary_issurance = genesis_epoch_ext
```

Comment:
Block 1 ~ 11 only creates cellbase outputs for primary issuance, we don't need to add `10*g2` here

Reply:
Since we're going to remove it soon, it's OK to just keep it.

Reply:
Sorry, not sure I get it: what are we going to remove soon?

Reply:
Remove the extra 10 blocks rewards.